### DISCUSSION TITLE
Overlap a transparent image into a mockup

### DISCUSSION THREAD

**Discussion by fraser100** [ImageMagick/ImageMagick#5454] [General]
I’m trying to overlap one image over a wall art mockup. One image has been rotated and resized to fit inside the frame in the mockup. After rotating the image, white triangles appear around it. How do I make the image transparent?

![Mockup](https://user-images.githubusercontent.com/111435974/185130376-f93f991f-c58f-47fe-bc65-b076a2d7861e.jpg)

**Comment by snibgo** [comment]
What version of IM, on what platform?

Please show the command you used. Otherwise, we have to guess.

Perhaps you need `-background transparent` (or `-background none` which is the same thing) before rotating.

**Reply by fraser100** [reply]
ImageMagick 7.1.0-45 for Windows

That was a good guess!! background transparent worked. Thanks.

The image is not fitting in the frame correctly so been messing about with Bilinear Distortion. Not sure if this is the correct setting to use though managed to fit the image into the frame using Skew on Photoshop.

I’m receiving the error “invalid list of numbers '-distort' ''0,0' at CLI arg 8” after entering the below.

magick input.png -alpha set -virtual-pixel transparent -interpolate Spline -distort BilinearForward '0,0 26,0 128,0 114,23 128,128 128,100 0,128 0,123' out.png

Any help is much appreciated

**Reply by fmw42** [reply]
use +distort perspective after finding the 4 corners of the image and 4 corners where you want to place it in the background

**Reply by fraser100** [reply]
I think I've found 4 corners of the 1651 x 1101 image. However, I'm still receiving the same error.

magick input.png -alpha set -virtual-pixel transparent +distort perspective '0,0,0,0  0,1100,0,1000  1650,0,1750,1650  1650,1100,1750,1000' output.png

**Reply by fmw42** [reply]
You probably do not have the correct corner coordinates for the area in the frame that you want to put the image.  The 4 corners of the input image are specified by the dimensions of the image. The other set is the 4 corners of the part of the inner frame of the background image into which you want to fit the input image. You should measure them using an interactive tool such as GIMP for example.

**Reply by fmw42** [reply]
See https://imagemagick.org/Usage/distorts/#control_points for control point format.

**Reply by fraser100** [reply]
Distort perspective worked perfectly!! Thanks so much for the help. How do I composite an image onto a canvas? Instead of into frames.

**Comment by snibgo** [comment]
You are using Windows, so use double-quotes " not single quotes '

Or replace the spaces between the numbers with commas.

**Reply by fraser100** [reply]
Worked after changing to double-quotes. Thanks.
